### PR TITLE
Add ProductThumbnail to Checkout

### PR DESCRIPTION
**Pull request by jkeohan**
### Background

Resolves https://github.com/Shopify/checkout-web/issues/25236
- Adds ProductThumbnail component to Checkout surface

### Solution

Copied the types and docs from private packages

### 🎩

[https://shopify-dev.shopify-dev-product-thumbnail.joe-keohan.eu.spin.dev/docs/api/checkout-ui-extensions/unstable/components/media/productthumbnail](https://shopify-dev.shopify-dev-product-thumbnail.joe-keohan.eu.spin.dev/docs/api/checkout-ui-extensions/unstable/components/media/productthumbnail)

### Checklist

- [ ] I have :tophat:'d these changes
- [x] I have updated relevant documentation

### CODE REVIEW

File: packages/ui-extensions/src/surfaces/checkout/components/ProductThumbnail/ProductThumbnail.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+   */
+  alt?: string;
+
+  /**
+   * A label that describes the purpose or contents of the icon.
+   *
+   * When set, it will be announced to users using assistive technologies and will provide them with more context.
+   * This should only be used if the icon requires an alternative internationalised label
+   * or if it is otherwise inappropriate to make use of the default label included with the icon.
+   */
```
[the remark this answers is not in the document]
Comment (jkeohan) [reply packages/ui-extensions/src/surfaces/checkout/components/ProductThumbnail/ProductThumbnail.ts:21]: Updated the API's for both `accessibilityLabel` and `alt` description text in [this PR](https://github.com/Shopify/checkout-web/pull/26056)

File: packages/ui-extensions/src/surfaces/checkout/components/ProductThumbnail/ProductThumbnail.doc.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      ],
+    },
+  },
+  subSections: [
+    {
+      type: 'Generic',
+      anchorLink: 'best-practices',
+      title: 'Best Practices',
+      sectionContent:
+        '**High-quality images and consistent aspect ratio**\n\n- Use optimized product images that ensure visual clarity and loading speed. Maintain a consistent aspect ratio for product thumbnails to avoid distortion or stretching of the images.\n\n**Consistent visual style and appropriate sizes**\n\n- Keep a consistent visual style for product thumbnails throughout your store. Use appropriate size for product thumbnails depending on the layout and use case. This consistency helps buyers recognize and associate the thumbnails with your product offerings.\n\n**Accessibility considerations**\n\n- Ensure that product thumbnails are accessible to users with disabilities. Provide alternative text (alt text) for screen readers and ensure that the thumbnails can be navigated using keyboard controls.',
```
[the remark this answers is not in the document]
Comment (jkeohan) [reply packages/ui-extensions/src/surfaces/checkout/components/ProductThumbnail/ProductThumbnail.doc.ts:44]: @nayeobkim do we need the additional text?  any thoughts on just rewording it: `Provide alternative text (alt text) for screen readers and ensures that product thumbnails are accessible to users with disabilities`.

File: packages/ui-extensions/docs/surfaces/checkout/screenshots/productthumbnail-thumbnail.png
[the remark this answers is not in the document]
Comment (jkeohan) [reply packages/ui-extensions/docs/surfaces/checkout/screenshots/productthumbnail-thumbnail.png:1]: @rockytori Should we include the Image preview in this PR or perhaps another that combines it with updating the icons preview image to show the new icons.

[the remark this answers is not in the document]
Comment (lsit) [reply packages/ui-extensions/docs/surfaces/checkout/screenshots/productthumbnail-thumbnail.png:1]: I think we should update the Image and Icon components separately

[the remark this answers is not in the document]
Comment (rockytori) [reply packages/ui-extensions/docs/surfaces/checkout/screenshots/productthumbnail-thumbnail.png:1]: Agreed with @lsit

[the remark this answers is not in the document]
Comment (nayeobkim) [reply packages/ui-extensions/docs/surfaces/checkout/screenshots/productthumbnail-thumbnail.png:1]: Everything else looks good to me

File: packages/ui-extensions/src/surfaces/checkout/components/ProductThumbnail/ProductThumbnail.doc.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (nayeobkim) [reply packages/ui-extensions/src/surfaces/checkout/components/ProductThumbnail/ProductThumbnail.doc.ts:44]: @jkeohan I think we can update the copy to

> Ensure product thumbnails are accessible with descriptive alternative text (alt text).

as @rockytori suggested!

**Review by jffortier — APPROVED** [review]

**Pull request merged**